assignment-1/Functions.java

```diff
@@ -0,0 +1,271 @@
+            File file = new File(csvFile);
+            FileReader fr = new FileReader(file);
+            BufferedReader br = new BufferedReader(fr);
+            String line = "";
+            String[] individualField;
+
+            //allPlayers
+
+            while ((line = br.readLine()) != null) {
+                while ((line = br.readLine()) != null) {
+                    individualField = line.split(delimiter);
+                    String playerName = individualField[0];
+                    String playerTeam = individualField[1];
+                    String playerRole = individualField[2];
+                    int playerMatches = Integer.parseInt(individualField[3]);
+                    int playerRuns = Integer.parseInt(individualField[4]);
+                    Double playerAverage = Double.parseDouble(individualField[5]);
+                    Double playerStrikeRate = Double.parseDouble(individualField[6]);
+                    int playerWickets = Integer.parseInt(individualField[7]);
+                    allPlayers.add(new Players(playerName, playerTeam, playerRole, playerMatches, playerRuns, playerAverage, playerStrikeRate, playerWickets));
+                }
+            }
+
+            //allTeams object
+
+            for (Players player : allPlayers) {
+                boolean flag = false;
+                String temp = player.getTeam();
+                for (Team t : allTeams) {
+                    if (t.Name.equals(temp)) {
+                        flag = true;
+                        t.players.add(player);
+                        t.Name = temp;
+                        break;
+                    }
+                }
+                if (!flag) {
+                    ArrayList<Players> play = new ArrayList<Players>();
+                    play.add(new Players(player.getName(), player.getTeam(), player.getRole(), player.getMatches(), player.getRuns(), player.getAverage(), player.getStrikeRate(), player.getWickets()));
+                    allTeams.add(new Team(temp, play));
+                }
+            }
+
+            br.close();
+        } catch (FileNotFoundException fnfe) {
+            System.out.println(fnfe.getMessage());
+        } catch (IOException ioe) {
+            System.out.println(ioe.getMessage());
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    //Second Task
+    public static void matchFixture() throws FileNotFoundException {
+        try {
+            PrintWriter pw = new PrintWriter(new File("fixture.csv"));
+            StringBuilder sb = new StringBuilder();
+
+            sb.append("Sr.No");
+            sb.append(",");
+            sb.append("Guest Team");
+            sb.append(",");
+            sb.append(("Home Team"));
+            sb.append(",");
+            sb.append("Venue-Ground");
+            sb.append("\n");
+            Set<String> hash_Set = new HashSet<String>();
+            for (Players t : allPlayers) {
```

Comment:
1. could have retrieved this in an elegant way using filters. or by storing a list of teams while reading the main csv file.
2. or, the better option would have been to use the allteam variable to get the team names.